ads/google/a4a/utils.js

```diff
@@ -480,3 +488,24 @@ export function extractAmpAnalyticsConfig(
   }
   return null;
 }
+
+/**
+ * Add new experiment IDs to a (possibly empty) existing set of experiment IDs.
+ * The {@code currentIdString} may be {@code null} or {@code ''}, but if it is
+ * populated, it must contain a comma-separated list of integer experiment IDs
+ * (per {@code parseExperimentIds()}).  Returns the new set of IDs, encoded
+ * as a comma-separated list.  Does not de-duplicate ID entries.
+ *
+ * @param {!Array<string>} newIds IDs to merge in. Should contain stringified
+ *     integer (base 10) experiment IDs.
+ * @param {?string} currentIdString  If present, a string containing a
+ *   comma-separated list of integer experiment IDs.
+ * @returns {string}  New experiment list string, including newId iff it is
+ *   a valid (integer) experiment ID.
+ * @see parseExperimentIds, validateExperimentIds
+ */
+export function mergeExperimentIds(newIds, currentIdString) {
+  const newIdString = newIds.filter(newId => Number(newId)).join(',');
+  const separator = currentIdString && newIdString ? ',' : '';
```

Comment:
This seems more readable to me...
currentIdString = currentIdString || '';
return currentIdString + (currentIdString && newIdString ? ',' : '')+newIdString;